File: Daily/sbgo/Leetcode275-H指数II.go
package sbgo

func hIndex(citations []int) int {
	l, r, n := 0, len(citations), len(citations)
	check := func(idx int) bool {
		return citations[n-idx] >= idx
	}
	for l < r {
		mid := (l + r + 1) >> 1
		if check(mid) {
			l = mid
		} else {
			r = mid - 1
		}
	}
	return l
}

File: Daily/PY/Leetcode275-H指数II.py
from typing import List
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        l,r = 0,len(citations)
        n = len(citations)
        def check(idx):
            return citations[n - idx] >= idx
        while l < r:
            mid = (l + r + 1) >> 1
            if(check(mid)): l = mid
            else: r = mid - 1
        return l